cms/tests/test_log_operations.py

```diff
@@ -0,0 +1,209 @@
+from django.contrib.admin.models import LogEntry, ADDITION, CHANGE, DELETION
+
+from cms.api import create_page
+from cms.test_utils.testcases import (
+    CMSTestCase, URL_CMS_PAGE_MOVE,
+    URL_CMS_PAGE_CHANGE, URL_CMS_PAGE_ADD,
+)
+from cms.models.pagemodel import Page
+from cms.forms.wizards import CreateCMSPageForm
+from cms.wizards.forms import step2_form_factory, WizardStep2BaseForm
+
+# Snippet to create wizard page taken from: test_wizards.py
+CreateCMSPageForm = step2_form_factory(
+    mixin_cls=WizardStep2BaseForm,
+    entry_form_class=CreateCMSPageForm,
+)
+
+
+class LogPageOperationsTests(CMSTestCase):
+
+    def setUp(self):
+        pass
+
+    def _assert_page_addition_log_created(self, page):
+
+        # Check to see if the page added log entry exists
+        self.assertEqual(1, LogEntry.objects.count())
+
+        log_entry = LogEntry.objects.all()[0]
+
+        # Check that the contents of the log message is correct
+        message = '[{"added": {}}]'
+        self.assertEqual(message, log_entry.change_message)
+        message = 'Added.'
+        self.assertEqual(message, log_entry.get_change_message())
```

Comment:
@czpython how far does the removal of LogEntry dependance need to go i.e.the test for get_change_message is specific to how the admin uses the multi lingual json objects in the change_message field. I sthis test redundant and does that also mean any auto generated messages are also redundant. I'm confused at just how far away this task needs to move away from the Admin LogEntry model.

Reply:
Good point. We don't need to test ``get_change_message``.